Dockerfile

```diff
@@ -1,30 +1,71 @@
-FROM haskell:8.6 AS build
-USER root
-# TODO: install Yices, too (and CVC4?, Boolector?)
+from debian:buster AS solvers
+
+# Install needed packages for building
 RUN apt-get update \
-    && apt-get install -y wget libncurses-dev unzip \
-    && wget https://github.com/Z3Prover/z3/releases/download/z3-4.7.1/z3-4.7.1-x64-debian-8.10.zip \
-    && unzip z3*.zip \
-    && mv z3-*/bin/z3 /usr/local/bin
+    && apt-get install -y curl cmake gcc g++ git libreadline-dev unzip
+RUN useradd -m user
+RUN install -d -o user -g user /solvers
+USER user
+WORKDIR /solvers
+RUN mkdir -p rootfs/usr/local/bin
+
+# Get Z3 4.8.8 from GitHub
+RUN curl -L https://github.com/Z3Prover/z3/releases/download/z3-4.8.8/z3-4.8.8-x64-ubuntu-16.04.zip --output z3.zip
+RUN unzip z3.zip
+RUN mv z3-*/bin/z3 rootfs/usr/local/bin
+
+# Build abc from GitHub. (Latest version.)
+RUN git clone https://github.com/berkeley-abc/abc.git
+RUN cd abc && make -j$(nproc)
+RUN cp abc/abc rootfs/usr/local/bin
+
+# Build Boolector release 3.2.1 from source
+RUN curl -L https://github.com/Boolector/boolector/archive/3.2.1.tar.gz | tar xz
+RUN cd boolector* && ./contrib/setup-lingeling.sh && ./contrib/setup-btor2tools.sh && ./configure.sh && cd build && make -j$(nproc)
+RUN cp boolector*/build/bin/boolector rootfs/usr/local/bin
+
+# Install Yices 2.6.2
+RUN curl -L https://yices.csl.sri.com/releases/2.6.2/yices-2.6.2-x86_64-pc-linux-gnu-static-gmp.tar.gz | tar xz
+RUN cp yices*/bin/yices-smt2 rootfs/usr/local/bin \
+    && cp yices*/bin/yices rootfs/usr/local/bin
+
+# Install CVC4 1.8
+RUN curl -L https://github.com/CVC4/CVC4/releases/download/1.8/cvc4-1.8-x86_64-linux-opt --output rootfs/usr/local/bin/cvc4
```

Comment:
Do we need to use the nightly version of CVC4 like we do in the cryptol Dockerfile?

Reply:
I'm not entirely sure. I'll leave it as-is for the moment, but we should look into that.